Add TypeScript support to src files

tbarri opened the pull request:
This commit consists of a series of changes that convert the src
JavaScript (JS) files to TypeScript (TS). The conversion process
involved renaming all files, updating import and export statements,
and finally backporting various types.

Babel was previously used to transcompile files directly on the src
folder. It is now instead redirected to operate on the output of the TSC
compiler.

PR: https://github.com/identity-com/credential-commons/pull/210
ISSUE: IDCOM-2324

---

- Add TypeScript support to src files
- Add YARN type dependencies
- Switch eslint to typescript
- Convert src/constants/*
- Convert src/errors/*
- Convert src/lib/stringUtils
- Convert src/creds/crypto
- Convert src/lib/did
- Convert lib/signerVerifier
- Convert src/SecureRandom
- Convert src/timeHelper
- Convert src/AggregationHandler
- Convert src/creds/claimModel
- Convert src/creds/__mocks__/definitions
- Convert src/claim/__mocks__/definitions
- Convert src/creds/CredentialSignerVerifier
- Convert src/services/__mocks__/httpService
- Convert src/isValidGlobalIdentifier
- Convert src/logger
- Convert src/services/config
- Convert src/services/MiniCryptoManagerImpl
- Convert src/claim/definitions
- Convert src/creds/definitions
- Convert src/services/httpService
- Convert src/claim/Claim
- Convert src/schemas/jsonSchema
- Convert src/schemas/jsonSchema/loaders/cvc
- Convert src/uca/UCA
- Convert src/creds/CvcMerkleProof
- Convert src/isClaimRelated
- Convert src/services/anchorService
- Convert src/creds/VerifiableCredential
- Convert src/creds/VerifiableCredentialProxy
- Convert src/services/DefaultAnchorServiceImpl
- Convert src/services/DummyAnchorServiceImpl
- Convert src/services/index
- Convert src/index.ts
- Address linting errors
- Restore default import statements
- Run formatter

tbarri committed:
Add TypeScript support to src files

This commit consists of a series of changes that convert the src
JavaScript (JS) files to TypeScript (TS). The conversion process
involved renaming all files, updating import and export statements,
and finally backporting various types.

Babel was previously used to transcompile files directly on the src
folder. It is now instead redirected to operate on the output of the TSC
compiler.

PR: https://github.com/identity-com/credential-commons/pull/210
ISSUE: IDCOM-2324

Reviewed-by: Julian Spring <springjulian@gmail.com>
Reviewed-by: William Brooks <william@identity.org>
Signed-off-by: Tighe Barris <4658684+tbarri@users.noreply.github.com>

JulianSpring commented on __test__/lib/util/did.js:
```diff
@@ -1,6 +1,6 @@
 const nacl = require('tweetnacl');
 const bs58 = require('bs58');
-const didUtil = require('../../../src/lib/did');
+const didUtil = require('../../../dist/ts-generated/lib/did');
```
It might be nice to add an alias, in our tsconfig to the top folder so we don't have to backtrack like this.
```
"paths": {
      "@/*": ["./src/*"]
    }
```

`@/dist/ts-generated/lib/did`

JulianSpring replied:
Nice thank you!

tbarri committed:
Update eslint related packages

This commit updates neighbouring dependencies of eslint. Previous
version were dated and causing builds to fail after the introduction of
commit cf1b086 ("Add TypeScript support to src files")

PR: https://github.com/identity-com/credential-commons/pull/210
ISSUE: IDCOM-2324

Reviewed-by: Julian Spring <springjulian@gmail.com>
Signed-off-by: Tighe Barris <4658684+tbarri@users.noreply.github.com>
